### PR TITLE
stats: if the stats in kv store is invalid, we will replace it with peusdo one.

### PR DESCRIPTION
**Pull request by hanfei1991**
If tidb1 analyze a table, and then do ddl operation. After that if a new tidb2 is on, it will read the stats with old version and panic.

@shenli @coocood @zimulala PTAL

### CODE REVIEW

**Comment by hanfei1991** [comment]
@lamxTyler PTAL

**Comment by coocood** [comment]
Need test.

File: plan/statscache/statscache.go [hunk 1]
```diff
@@ -68,7 +68,8 @@ func (h *Handle) Update(m *meta.Meta, is infoschema.InfoSchema) error {
 			// Error is not nil may mean that there are some ddl changes on this table, so the origin
 			// statistics can not be used any more, we give it a nil one.
```
Comment (lamxTyler) [inline plan/statscache/statscache.go]: The comment should be updated, it is a pseudo one.

**Comment by lamxTyler** [comment]
I think the bug is on line https://github.com/pingcap/tidb/blob/ac7418cee97568ff01af7eeb39fd4a17d25a3e4c/plan/statscache/statscache.go#L93.
Here check the `stats == nil` redundant, we should check `stats.tbl==nil`.

**Comment by hanfei1991** [comment]
@lamxTyler Well, you are right, but it's more safe to avoid to use nil pointer.

File: plan/statscache/statscache.go [hunk 2]
```diff
@@ -115,7 +122,7 @@ func SetStatisticsTableCache(id int64, statsTbl *statistics.Table, version uint6
 		return
 	}
 	if stats.version >= version {
-		return
+		//		return
```
Comment (coocood) [inline plan/statscache/statscache.go]: Why comment it out?

File: plan/statscache/statscache.go [hunk 3]
```diff
@@ -34,10 +34,17 @@ type statsInfo struct {
 
 // Handle can update stats info periodically.
 type Handle struct {
-	ctx         context.Context
+	ctx context.Context
+	// LastVersion stands for the last time we update stats. We export it only for test.
```
Comment (coocood) [inline plan/statscache/statscache.go]: not exported, update the comment

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
@lamxTyler PTAL

**Comment by lamxTyler** [comment]
LGTM

**Pull request merged**